dex/tests/test.sh

```diff
@@ -1,39 +1,42 @@
 #!/bin/sh
+#/ Usage: test.sh <pkg_ident>
+#/
+#/ Example: test.sh core/dex/2.17.0/20190531065119
+#/
 
-TESTDIR="$(dirname "${0}")"
-PLANDIR="$(dirname "${TESTDIR}")"
-SKIPBUILD=${SKIPBUILD:-0}
+set -euo pipefail
+
+if [[ -z "${1:-}" ]]; then
+  grep '^#/' < "${0}" | cut -c4-
+  exit 1
+fi
 
+TESTDIR="$(dirname "${0}")"
 source "${TESTDIR}/helpers.bash"
 
-hab pkg install core/bats --binlink
+export TEST_PKG_IDENT="${1}"
+hab pkg install "${TEST_PKG_IDENT}"
 
 hab pkg install core/busybox-static
 hab pkg binlink core/busybox-static nc
-hab pkg install -b core/jq-static core/curl
+hab pkg install core/bats core/curl core/jq-static --binlink
 
-# Wait for supervisor to start
-echo "Waiting for supervisor to start"
-wait_listen tcp 9632 30
+hab sup term
+while hab sup status 2>/dev/null; do
+  echo "Waiting for supervisor to terminate"
+  sleep 1
+done
 
-source "${PLANDIR}/plan.sh"
-# Unload the service if its already loaded.
-hab svc unload "${HAB_ORIGIN}/${pkg_name}"
+hab sup run &
 
-if [ "${SKIPBUILD}" -eq 0 ]; then
-  set -e
-  pushd "${PLANDIR}" > /dev/null
-  build
-  popd > /dev/null
-  set +e
-fi
-
-source "${PLANDIR}/results/last_build.env"
-hab pkg install "${PLANDIR}/results/${pkg_artifact}" --binlink --force
-hab svc load "${pkg_ident}"
+echo "Waiting for supervisor to start (30s)"
+wait_listen tcp 9632 30
+hab svc load "${TEST_PKG_IDENT}"
 
 # Wait for 5 seconds on first check, to ensure service is up.
-echo "Waiting for dex to start (5s)"
-wait_listen tcp 5556 5
+echo "Waiting for service to start (5s)"
+wait_listen tcp 8181 5
```

Comment:
It looks like dex is starting on 5556 in CI, leading to a timeout waiting for it to come up. Should it be listening on 8181?

Reply:
ha I'm an idiot. of course. thanks